### PR TITLE
Empty directory creation on invalid model file path fixed/ added '/' to endpoints

### PR DESCRIPTION
**Pull request by vrakesh**
*Issue #, if available:*
#320 
#268 
*Description of changes:*
During handling of the model file exception ensure the model_Dir is deleted, also added unit test for the functionality.
Also added a default root '/' path endpoint which reuses the ping functionality, this should be a good resolution for #268 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by lupesko** [comment]
@vrakesh it seems you have a single PR that addresses multiple issues. This is not ideal since it makes the review process slower and less focuses.
It is better to have multiple small PRs, each specific to a given issue.

File: mms/model_loader.py [hunk 1]
```diff
@@ -80,8 +80,10 @@ def _download_and_extract(model_location, path=None, overwrite=False):
             if '.model' in model_file:
                 _extract_zip(model_file, model_dir)
         except Exception as e:
-            raise Exception('Failed to open model file %s for model %s. Stacktrace: %s'
-                            % (model_file, model_file_prefix , e))
+            #Remove the directory being created if invalid model specified
```
Comment (lupesko) [inline mms/model_loader.py]: nit: space after #

File: mms/serving_frontend.py [hunk 2]
```diff
@@ -182,20 +182,20 @@ def add_endpoint(self, api_definition, callback, **kwargs):
...
 
         Parameters
         ----------
         host : str
-            Host that server will use 
+            Host that server will use
 
         port: int
-            Host that server will use 
+            Host that server will use
```
Comment (lupesko) [inline mms/serving_frontend.py]: Should be Port not Host

File: mms/serving_frontend.py [hunk 3]
```diff
@@ -182,20 +182,20 @@ def add_endpoint(self, api_definition, callback, **kwargs):
...
         """
         modelservices = self.service_manager.get_loaded_modelservices()
         # TODO: not hardcode host:port
```
Comment (lupesko) [inline mms/serving_frontend.py]: Is this issue still relevant, or should we remove the comment?
If still relevant please file an issue and add the link here.

File: mms/serving_frontend.py [hunk 4]
```diff
@@ -346,18 +346,44 @@ def setup_openapi_endpoints(self, host, port):
...
                     }
                 }
-                
+
             }
         }
         self.openapi_endpoints['paths'].update(api_description_api)
         self.add_endpoint(api_description_api, self.api_description)
 
+        #4. root endpoint (A secondary way  to ping functionality)
```
Comment (lupesko) [inline mms/serving_frontend.py]: Can we please add tests to check the new endpoint?

**Comment by vrakesh** [comment]
@lupesko Sure, going forward will ensure, each PR is restricted to a single issue. Will work on the requested changes

File: mms/tests/unit_tests/test_model_loader.py [hunk 5]
```diff
@@ -16,3 +16,13 @@ def test_onnx_fails_fast():
...
+
+def test_invalid_model_path_input():
+    """
+    Test to ensure that folder being created is removed,
+    when path is invalid
+    """
+    models = {'squeezenet_v1': 'invalid_model_file_path.model'}
+    with pytest.raises(Exception) as e:
+        ModelLoader.load(models)
+    assert not os.path.exists('invalid_model_file_path')
```
Comment (vdantu) [inline mms/tests/unit_tests/test_model_loader.py]: Could you add one new line without any space here?

**Comment by vrakesh** [comment]
@lupesko @vdantu added and pushed all requested changes

**Pull request merged**